File: py_homework/4.3/04.py
a = [[1, 2, 3], [4, 5, 6], [7, 8, 9]]
b = [[9, 8, 7], [6, 5, 4], [3, 2, 1]]

c = [[a[i][j] + b[i][j] for j in range(3)] for i in range(3)]
print(c)

d = [[a[i][j] - b[i][j] for j in range(3)] for i in range(3)]
print(d)

e = [[sum(a[i][k] * b[k][j] for k in range(3)) for j in range(3)] for i in range(3)]
print(e)

f = sum(a[i][i] for i in range(3))
print(f)

File: Luohgu/【算法1-5】贪心/04.cpp
#include<bits/stdc++.h>
using namespace std;

int n;
long long sum;

int main(){
    ios::sync_with_stdio(false);
    cin.tie(nullptr);
    cin >> n;
    priority_queue<int, vector<int>, greater<int>> pq;
    for (int i = 0; i < n;++i){
        int w;
        cin >> w;
        pq.push(w);
    }
    for (int i = 1; i < n;++i){
        int m1 = pq.top();
        pq.pop();
        int m2 = pq.top();
        pq.pop();
        int m = m1 + m2;
        pq.push(m);
        sum += m;
    }
    cout << sum << "\n";
    return 0;
}